extensions/amp-analytics/0.1/events.js

```diff
@@ -400,6 +409,151 @@ export class ClickEventTracker extends EventTracker {
   }
 }
 
+/**
+ * Tracks scroll events.
+ */
+export class ScrollEventTracker extends EventTracker {
+
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {!./analytics-root.AnalyticsRoot} root */
+    this.root_ = root;
+
+    /** @private {function(!Object)|null} */
+    this.scrollHandler_ = null;
+  }
+
+  /** @override */
+  dispose() {
+    if (this.scrollHandler_ !== null) {
+      this.root_.getScrollManager()
+          .removeScrollHandler(this.scrollHandler_);
+      this.scrollHandler_ = null;
+    }
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    if (!config['scrollSpec']) {
+      user().error(TAG, 'Missing scrollSpec on scroll trigger.');
+      return NO_UNLISTEN;
+    }
+
+    if (!Array.isArray(config['scrollSpec']['verticalBoundaries']) &&
+      !Array.isArray(config['scrollSpec']['horizontalBoundaries'])) {
+      user().error(TAG, 'Boundaries are required for the scroll ' +
+        'trigger to work.');
+      return NO_UNLISTEN;
+    }
+
+    this.scrollHandler_ = this.getScrollHandler_(config, listener);
+
+    return this.root_.getScrollManager()
+        .addScrollHandler(this.scrollHandler_);
+  }
+
+  /**
+   * Function to return a Scroll Handler Function instance
+   * @param {!JsonObject} config
+   * @param {function(!AnalyticsEvent)} listener
+   * @return {function(!Object)}
+   * @private
+   */
+  getScrollHandler_(config, listener) {
```

Comment:
can we have a scrollHandler method instead of a method to return a scroll handler.